### DISCUSSION TITLE
After boot client can't connect to ID server after first login macOS (Sequioa 15)

### DISCUSSION THREAD

**Discussion by selimrecep** [rustdesk/rustdesk#10034] [General]
### Bug Description

RustDesk client seems to not be able to connect after login that is after a boot, on a M4 mac mini (Sequioa 15)

### How to Reproduce

0- Setup RustDesk with daemon and start on boot enabled
1- Reboot device, either via power down or reboot
2- Remote connect via RustDesk from another client (Linux or android), connects fine, login screen shows up
3- Enter password on login screen via RustDesk client
4- Right after hitting Enter key (logging in successfully), client disconnects
5- Client can't connect to MacOS client anymore because MacOS client can't connect to ID server (this is confirmed by trying to connect to another client on MacOS client, any id works, and gives message below):
**Failed to connect to XXX.XXX.XXX.XXX:21116: Please try later** (The IP is a local address, in same subnet)
6- In order to fix it you have to either keep uninstalling the daemon and install it few times (not sure what fixes this exactly this way)
7-   or, I have found a better way to fix it, that is through System Settings > Privacy & Security > Local Network > **Switch off and on** RustDesk. Now MacOS client can connect to the ID server.

### Expected Behavior

0- Setup RustDesk with daemon and start on boot enabled
1- Reboot device, either via power down or reboot
2- Remote connect via RustDesk from another client (Linux or android), connects fine, login screen shows up
3- Enter password on login screen via RustDesk client
4- Right after hitting Enter key (logging in successfully), client **should NOT disconnect** and still be accessible for remote connections.

### Operating system(s) on local (controlling) side and remote (controlled) side

Ubuntu 22.04 -> MacOS Sequioa 15
Android -> MacOS Sequioa 15
 MacOS Sequioa 15 -> Anything (as it cant connect to id server)

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.2 -> 1.3.2

### Screenshots
(Scraped from MacOS client)
![Screenshot 2024-11-24 at 01 51 43](https://github.com/user-attachments/assets/ffb12752-b830-453b-b52c-440767674ea1)

### Additional Context
I have found something related to this setting on this link: https://mjtsai.com/blog/2024/10/02/local-network-privacy-on-sequoia/ 
I am also kinda new to MacOS so not exactly sure what's going on here but before login RustDesk can connect to ID server just fine, but somehow after login it needs the workaround I wrote above to get it accessible again. Maybe before login daemon somehow runs as root but after that something gets broken?
P.S.: I think in order to reproduce the IP address needs to be in same subnet within LAN connection (or Wi-Fi)

I have tried **rustdesk-1.3.2-aarch64.dmg** (last release) and **rustdesk-1.3.3-aarch64.dmg** (last beta) versions 

_P.S.2: There was also another bug that could be tracked in another issue, that is after a while after a remote connections is closed, the mac mini goes to sleep mode (it is not a full sleep i can still ssh into) and i can't connect to the mac anymore, when i SSH into the device and do nothing else, I can now access to the macos client from my linux client, i fixed this by disabling going to sleep as a workaround, but now maybe this issue could be related to this issue i am submitting too...)_

**Comment by selimrecep** [comment]
Okay I think the issue might relate to OS instead of rustdesk, a similar issue is reported here: https://github.com/deskflow/deskflow/issues/7895

**Reply by alan-roe** [reply]
Anydesk are recommending not to upgrade to Sequoia as it breaks the ability to connect from the first login screen. Definitely appears to be a change in the OS, remains to be seen what can be done about it.
